adapters/repos/db/aggregator/grouper.go

```diff
@@ -151,20 +151,32 @@ func (g *grouper) addElementById(s *models.PropertySchema, docID uint64) error {
 }
 
 func (g *grouper) addItem(item interface{}, docID uint64) {
-	ids := g.values[item]
-	ids = append(ids, docID)
-	g.values[item] = ids
+	idsMap, ok := g.values[item]
+	if !ok {
+		idsMap = map[uint64]struct{}{}
+	}
+	idsMap[docID] = struct{}{}
+	g.values[item] = idsMap
 }
 
 func (g *grouper) aggregateAndSelect() ([]group, error) {
-	for value, ids := range g.values {
+	for value, idsMap := range g.values {
+		count := len(idsMap)
+		ids := make([]uint64, count)
+
+		i := 0
+		for id := range idsMap {
+			ids[i] = id
+			i++
```

Comment:
where is `i` variable used? and what's the purpose of this for loop?

Reply:
`ids[i] = id`
docID are stored as keys in the map, to keep them unique. loop is meant to convert keys of map into array